docs/02-app/02-api-reference/05-next-config-js/typedEnv.mdx

```diff
@@ -0,0 +1,17 @@
+---
+title: typedEnv
+nav_title: typedEnv
+description: Enable type-checking and autocompletion for environment variables.
+---
+
+Enable type-checking and autocompletion for environment variables. You can opt out by setting the `typedEnv` value to false. This feature requires using TypeScript in your project.
+
+```js filename="next.config.js"
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  // enabled by default, opt out by setting to false
+  typedEnv: false,
```

Comment:
Let's give this at least a non-abbreviated name for readability. It's only typed once so the name can be even more descriptive.